### PR TITLE
Add basic usage example

### PR DESCRIPTION
**Pull request by BarbDMC**
This PR adds a basic usage example to the project, the code gives an example of how to generate a log file.
Solves #38 

As part of [Stack Builders Inc.](https://www.stackbuilders.com/) Hacktoberfest 2023 event.

### CODE REVIEW

File: examples/logger/app/Main.hs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        logMessage logFile message
+        -- Print the message to the console for visibility.
+        putStrLn $ "Logged: " <> T.unpack message
+        -- Pause the execution for 1 second.
+        threadDelay 1000000
+
+main :: IO ()
+main = do
+    -- Clear the contents of the log file by writing an empty Text to it.
+    TIO.writeFile logFilePath ""
```
Comment (dbalseiro) [inline examples/logger/app/Main.hs:37]: consider
```Haskell
TIO.writeFile logFilePath mempty
```

or 
```Haskell
writeFile logFilePath "" -- you'll not get significant improvement by using Text here
```

File: examples/logger/app/Main.hs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    threadDelay 12000000
+
+    -- Read the contents of the log file into a Text variable.
+    logContents <- TIO.readFile logFilePath
+
+    -- Print a header indicating the log file contents.
+    putStrLn "Log File Contents:"
+
+    -- Print the log file contents to the console.
+    TIO.putStrLn logContents
```
Comment (dbalseiro) [inline examples/logger/app/Main.hs:52]: Consider
```Haskell
TIO.readFie logFilePath >>= TIO.putStrLn
```

**Review by dbalseiro — COMMENTED** [review]
Good Job @BarbDMC! I left a couple of comments but it looks good to me so far

**Comment by DavidMazarro** [comment]
Thanks for the contribution! Remember to add the following line in the PRs description so that the contribution can be considered valid for Stack Builders' Hacktoberfest 2023 event:

> As part of [Stack Builders Inc.](https://www.stackbuilders.com/) Hacktoberfest 2023 event.

File: examples/logger/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+.stack-work/
+*~
```
Comment (Alex0jk) [inline examples/logger/.gitignore:2]: new line at the end of file

File: examples/logger/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Changelog for `logger-example`
```
Comment (Alex0jk) [inline examples/logger/CHANGELOG.md:1]: Do we need this file?

File: examples/logger/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+# logger-example
```
Comment (Alex0jk) [inline examples/logger/README.md:1]: Lets add a good description of what the example does here

File: examples/logger/package.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    main:                Main.hs
+    source-dirs:         app
+    ghc-options:
+    - -threaded
+    - -rtsopts
+    - -with-rtsopts=-N
+    dependencies:
+    - logger-example
+
+tests:
```
Comment (Alex0jk) [inline examples/logger/package.yaml:53]: I do not think we need this as there are no tests

File: examples/logger/src/Lib.hs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+module Lib
```
Comment (Alex0jk) [inline examples/logger/src/Lib.hs:1]: Are we using this module?

File: examples/logger/CHANGELOG.md (hunk 4 above)
Comment (BarbDMC) [reply examples/logger/CHANGELOG.md:1]: No, we don't need it. Let me remove this file.

File: examples/logger/package.yaml (hunk 6 above)
Comment (BarbDMC) [reply examples/logger/package.yaml]: You're right, I removed those test dependencies.

File: examples/logger/src/Lib.hs (hunk 7 above)
Comment (BarbDMC) [reply examples/logger/src/Lib.hs]: No, I'll delete that file.

File: examples/logger/app/Main.hs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        putStrLn $ "Logged: " <> message
+        threadDelay 1000000
+
+main :: IO ()
+main = do
+    TIO.writeFile logFilePath mempty
+
+    _ <- forkIO (simulateLoggingProcess logFilePath)
+
+    threadDelay 12000000
```
[the remark this answers is not in the document]
Comment (BarbDMC) [reply examples/logger/app/Main.hs]: Done!

File: examples/logger/README.md (hunk 5 above)
Comment (BarbDMC) [reply examples/logger/README.md:1]: I added a description. Let me know if the new description is good or if it needs to be more extensive.

**Review by Alex0jk — APPROVED** [review]

**Pull request merged**